src/phpDocumentor/Reflection/Php/File.php

```diff
@@ -38,35 +40,34 @@
     private $source = null;
 
     /** @var Fqsen[] $namespaces */
-    private $namespaces = array();
+    private $namespaces = [];
 
     /** @var string[] $includes */
-    private $includes = array();
+    private $includes = [];
 
     /** @var Function_[] $functions */
-    private $functions = array();
+    private $functions = [];
 
     /** @var Constant[] $constants */
-    private $constants = array();
+    private $constants = [];
 
     /** @var Class_[] $classes */
-    private $classes = array();
+    private $classes = [];
 
     /** @var Interface_[] $interfaces */
-    private $interfaces = array();
+    private $interfaces = [];
 
     /** @var Trait_[] $traits */
-    private $traits = array();
+    private $traits = [];
 
     /**
      * Initializes a new file descriptor with the given hash of its contents.
      *
+     *
      * @param string $hash An MD5 hash of the contents if this file.
-     * @param string $path
      * @param string|null $source
-     * @param DocBlock $docBlock
      */
-    public function __construct($hash, $path, $source = null, DocBlock $docBlock = null)
+    public function __construct(string $hash, string $path, string $source = null, DocBlock $docBlock = null)
```

Comment:
`$source` should be `?string` a nullable string the default value will still be `null`